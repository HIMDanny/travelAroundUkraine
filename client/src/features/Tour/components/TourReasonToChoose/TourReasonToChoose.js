import React from 'react';
import { Paper as MuiPaper, styled, Typography, Box } from '@mui/material';
import PropTypes from 'prop-types';

const Paper = styled((props) => <MuiPaper elevation={0} {...props} />)(({ theme }) => ({
  border: `1px solid ${theme.palette.divider}`,
  maxWidth: '190px',
  padding: '30px 15px',
  boxSizing: 'border-box',
  boxShadow: '0px 0px 15px rgba(0, 0, 0, 0.2)',
  position: 'relative',

  '&:before': {
    content: '""',
    position: 'absolute',
    top: '-15px',
    left: '-15px',
    width: '57px',
    height: '57px',
    backgroundColor: '#FFC700',
    borderRadius: '50%',
    zIndex: '0',
  },
}));

const BackgroundBox = styled(Box)({
  position: 'absolute',
  inset: 0,
  background: 'rgba(255, 255, 255, 0.2)',
  backdropFilter: 'blur(10px)',
  borderRadius: 10,

  zIndex: '0',
});

const TourReasonToChoose = ({ number, description }) => (
  <Paper>
    <BackgroundBox />
    <Box sx={{ position: 'relative' }}>
      <Typography color="primary" fontSize={22} fontWeight={700}>
        {number}
      </Typography>
      <Typography margin="10px 0" fontWeight={700}>
        {description}
      </Typography>
    </Box>
  </Paper>
);

TourReasonToChoose.propTypes = {
  number: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
};

export default TourReasonToChoose;
