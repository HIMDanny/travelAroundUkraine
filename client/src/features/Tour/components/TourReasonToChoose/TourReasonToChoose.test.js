import { render } from '@testing-library/react';
import TourReasonToChoose from './TourReasonToChoose';

jest.mock('@mui/material/Paper', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);
jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);

describe('TourReasonToChoose snapshot testing', () => {
  test('should TourReasonToChoose match snapshot', () => {
    const { asFragment } = render(<TourReasonToChoose number={1} description="test" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
