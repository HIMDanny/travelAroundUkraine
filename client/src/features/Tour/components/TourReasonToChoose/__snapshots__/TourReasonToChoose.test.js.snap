// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TourReasonToChoose snapshot testing should TourReasonToChoose match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div />
    <div>
      <p>
        1
      </p>
      <p>
        test
      </p>
    </div>
  </div>
</DocumentFragment>
`;
